### PR TITLE
fix(ClientError): handle ClientErrors in DynamoDB and Directory Service

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Description

Handle ClientErrors in DynamoDB and Directory Service.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/directoryservice/directoryservice_service.py [hunk 1]
```diff
@@ -117,21 +118,27 @@ def __describe_event_topics__(self, regional_client):
...
+                            event_topics.append(
+                                EventTopics(
+                                    topic_arn=event_topic["TopicArn"],
+                                    topic_name=event_topic["TopicName"],
+                                    status=event_topic["Status"],
+                                    created_date_time=event_topic["CreatedDateTime"],
+                                )
                             )
+                        self.directories[directory.id].event_topics = event_topics
+                    except ClientError as error:
```
Comment (jfagoagas) [inline prowler/providers/aws/services/directoryservice/directoryservice_service.py]: Are we going to handle all `ClientError` this way? Should it be? 

The error was the following `ClientException[122]: An error occurred (ClientException) when calling the DescribeEventTopics operation: Operation is not supported for Shared MicrosoftAD directories.`

What if we look for `Operation is not supported for Shared MicrosoftAD directories.`? Also we have the `MicrosoftAD` type so we can play with that.

File: prowler/providers/aws/services/directoryservice/directoryservice_service.py [hunk 2]
```diff
@@ -146,25 +153,38 @@ def __list_certificates__(self, regional_client):
...
+                        if (
+                            error.response["Error"]["Code"]
+                            == "UnsupportedOperationException"
+                        ):
+                            logger.warning(
+                                f"{directory.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
                             )
-                    self.directories[directory.id].certificates = certificates
+                            continue
```
Comment (jfagoagas) [inline prowler/providers/aws/services/directoryservice/directoryservice_service.py]: Is this `continue` right? Should it be one `tab` to the left?

File: prowler/providers/aws/services/directoryservice/directoryservice_service.py [hunk 3]
```diff
@@ -179,22 +199,27 @@ def __get_snapshot_limits__(self, regional_client):
...
+                            ]["ManualSnapshotsCurrentCount"],
+                            manual_snapshots_limit=snapshot_limit["SnapshotLimits"][
+                                "ManualSnapshotsLimit"
+                            ],
+                            manual_snapshots_limit_reached=snapshot_limit[
+                                "SnapshotLimits"
+                            ]["ManualSnapshotsLimitReached"],
+                        )
+                    except ClientError as error:
+                        logger.warning(
```
Comment (jfagoagas) [inline prowler/providers/aws/services/directoryservice/directoryservice_service.py]: Are we going to handle all ClientError this way? Should it be?

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/providers/aws/services/directoryservice/directoryservice_service.py (hunk 1 above)
Comment (MrCloudSec) [reply prowler/providers/aws/services/directoryservice/directoryservice_service.py]: Done

File: prowler/providers/aws/services/directoryservice/directoryservice_service.py (hunk 2 above)
Comment (MrCloudSec) [reply prowler/providers/aws/services/directoryservice/directoryservice_service.py]: Changed

File: prowler/providers/aws/services/directoryservice/directoryservice_service.py (hunk 3 above)
Comment (MrCloudSec) [reply prowler/providers/aws/services/directoryservice/directoryservice_service.py]: Changed

**Review by jfagoagas — APPROVED** [review]
🚀

**Pull request merged**